pkg/http/rest/status.go

```diff
@@ -9,33 +9,35 @@ import (
 func HasValidContent(resp *http.Response) bool {
 	if resp != nil {
 		code := resp.StatusCode
-		return codeOK(code) && Is2xx(code) &&
+		return codeOK(code) && Is2xx(resp) &&
 			code != http.StatusNoContent &&
 			resp.Body != nil
 	}
 
 	return false
 }
 
-func Is4xx(code int) (bool, error) {
+func Is4xx(resp *http.Response) (bool, error) {
+	code := resp.StatusCode
 	if code >= http.StatusBadRequest && code <= 499 {
-		return true, errors.New(fmt.Sprintf("Received HTTP%d", code))
+		return true, errors.New(fmt.Sprintf("Received HTTP%d: %v", code, resp.Body))
 	}
 
 	return false, nil
 }
 
-func Is5xx(code int) (bool, error) {
+func Is5xx(resp *http.Response) (bool, error) {
+	code := resp.StatusCode
 	if code >= http.StatusInternalServerError && code <= 599 {
-		return true, errors.New(fmt.Sprintf("Received HTTP%d", code))
+		return true, errors.New(fmt.Sprintf("Received HTTP%d: %v", code, resp.Body))
 	}
 
 	return false, nil
 }
 
-func Is2xx(code int) bool {
-	return codeOK(code) &&
-		code >= http.StatusOK && code <= 299
+func Is2xx(resp *http.Response) bool {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [exported function Is2xx should have comment or be unexported](https://app.codacy.com/manual/TurnsCoffeeIntoScripts/jira-api-resource/pullRequest?prid=4777494)